Dart/src/com/jetbrains/lang/dart/ide/generation/DartGenerateAccessorHandler.java

```diff
@@ -25,25 +22,17 @@ protected BaseCreateMethodsFix createFix(DartClass dartClass) {
     return new CreateGetterSetterFix(dartClass, myStrategy);
   }
 
-  @Override
-  protected void collectCandidates(DartClass dartClass, List<DartComponent> candidates) {
-    final List<DartComponent> subComponents = DartResolveUtil.getNamedSubComponents(dartClass);
-
-    for (DartComponent dartComponent : subComponents) {
-      if (DartComponentType.typeOf(dartComponent) != DartComponentType.FIELD) continue;
-      if (dartComponent.isStatic()) continue;
-
-      if (!myStrategy.accept(dartComponent.getName(), subComponents)) continue;
-
-      candidates.add(dartComponent);
-    }
-  }
 
   @Override
-  protected void collectCandidates(Map<Pair<String, Boolean>, DartComponent> classMembersMap,
-                                   Map<Pair<String, Boolean>, DartComponent> superClassesMembersMap,
-                                   Map<Pair<String, Boolean>, DartComponent> superInterfacesMembersMap,
-                                   List<DartComponent> candidates) {
-    // ignore
+  protected void collectCandidates(final DartClass dartClass, List<DartComponent> candidates) {
+    candidates.addAll(ContainerUtil.findAll(computeClassMembersMap(dartClass).values(), new Condition<DartComponent>() {
+      @Override
+      public boolean value(DartComponent component) {
+        return DartComponentType.typeOf(component) == DartComponentType.FIELD &&
+               !component.isStatic() &&
+               myStrategy.accept(component.getName(), DartResolveUtil.getNamedSubComponents(dartClass));
```

Comment:
`DartResolveUtil.getNamedSubComponents(dartClass)` is too heavy to repeat it multiple times inside a filter. I'll extract it back.